Following test from XDB test model failed when running on Oracle 11gR2 (11.2.0.1)database.
Please note, this test passes on Oracle 11gR1 (11.1.0.7) and also
XDB tests uses OCI connection which requires usage of corresponding xdb.jar from the database. 
-------
[EL Finer]: 2009-11-05 14:13:32.85--DatabaseSessionImpl(27211574)--Thread(Thread[Thread-3,6,main])--Begin ExistsNodeWithJoinTest
[EL Finest]: 2009-11-05 14:13:32.863--DatabaseSessionImpl(27211574)--Thread(Thread[Thread-3,6,main])--Execute query ReadObjectQuery(referenceClass=Employee_XML )
[EL Fine]: 2009-11-05 14:13:32.9--DatabaseSessionImpl(27211574)--Connection(29812760)--Thread(Thread[Thread-3,6,main])--SELECT t1.ID, t1.FIRST_NAME, t1.LAST_NAME, t1.GENDER, t1.RESUME_XML, t1.PAYROLL_XML, t1.MANAGER_ID FROM EMPLOYEE_XML t0, EMPLOYEE_XML t1 WHERE ((existsNode(t0.RESUME_XML,?) > ?) AND (t0.ID = t1.MANAGER_ID))
	bind => [//education/degree, 0]
[EL Fine]: 2009-11-05 14:13:37.207--DatabaseSessionImpl(27211574)--Thread(Thread[Thread-3,6,main])--SELECT 1 FROM DUAL
[EL Info]: 2009-11-05 14:13:37.208--DatabaseSessionImpl(27211574)--Thread(Thread[Thread-3,6,main])--Communication failure detected when attempting to perform read query outside of a transaction. Attempting to retry query.

****************

Exception [EclipseLink-4002] (Eclipse Persistence Services - 1.2.0.qualifier): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: java.sql.SQLRecoverableException: Closed Connection
Error Code: 17008
Call: SELECT t1.ID, t1.FIRST_NAME, t1.LAST_NAME, t1.GENDER, t1.RESUME_XML, t1.PAYROLL_XML, t1.MANAGER_ID FROM EMPLOYEE_XML t0, EMPLOYEE_XML t1 WHERE ((existsNode(t0.RESUME_XML,?) > ?) AND (t0.ID = t1.MANAGER_ID))
    bind => [//education/degree, 0]
Query: ReadObjectQuery(referenceClass=Employee_XML sql="SELECT t1.ID, t1.FIRST_NAME, t1.LAST_NAME, t1.GENDER, t1.RESUME_XML, t1.PAYROLL_XML, t1.MANAGER_ID FROM EMPLOYEE_XML t0, EMPLOYEE_XML t1 WHERE ((existsNode(t0.RESUME_XML,?) > ?) AND (t0.ID = t1.MANAGER_ID))")
    at org.eclipse.persistence.exceptions.DatabaseException.sqlException(DatabaseException.java:333)
    at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.processExceptionForCommError(DatabaseAccessor.java:1411)
    at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.basicExecuteCall(DatabaseAccessor.java:677)
    at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.executeCall(DatabaseAccessor.java:528)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeCall(AbstractSession.java:914)
    at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:205)
    at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.executeCall(DatasourceCallQueryMechanism.java:191)
    at org.eclipse.persistence.internal.queries.DatasourceCallQueryMechanism.selectOneRow(DatasourceCallQueryMechanism.java:619)
    at org.eclipse.persistence.internal.queries.ExpressionQueryMechanism.selectOneRowFromTable(ExpressionQueryMechanism.java:2563)
    at org.eclipse.persistence.internal.queries.ExpressionQueryMechanism.selectOneRow(ExpressionQueryMechanism.java:2534)
    at org.eclipse.persistence.queries.ReadObjectQuery.executeObjectLevelReadQuery(ReadObjectQuery.java:431)
    at org.eclipse.persistence.queries.ObjectLevelReadQuery.executeDatabaseQuery(ObjectLevelReadQuery.java:997)
    at org.eclipse.persistence.queries.DatabaseQuery.execute(DatabaseQuery.java:670)
    at org.eclipse.persistence.queries.ObjectLevelReadQuery.execute(ObjectLevelReadQuery.java:958)
    at org.eclipse.persistence.queries.ReadObjectQuery.execute(ReadObjectQuery.java:399)
    at org.eclipse.persistence.internal.sessions.AbstractSession.internalExecuteQuery(AbstractSession.java:2322)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1225)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1299)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1299)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1299)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1299)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1207)
    at org.eclipse.persistence.internal.sessions.AbstractSession.executeQuery(AbstractSession.java:1167)
    at org.eclipse.persistence.internal.sessions.AbstractSession.readObject(AbstractSession.java:2730)
    at org.eclipse.persistence.testing.tests.xdb.ExistsNodeWithJoinTest.test(ExistsNodeWithJoinTest.java:36)
    at org.eclipse.persistence.testing.framework.TestCase.executeTest(TestCase.java:543)
    at org.eclipse.persistence.testing.framework.TestCase.execute(TestCase.java:154)
    at org.eclipse.persistence.testing.framework.TestExecutor.execute(TestExecutor.java:250)
    at org.eclipse.persistence.testing.framework.TestExecutor.runTest(TestExecutor.java:671)
    at org.eclipse.persistence.testing.framework.ui.SynchronizedTestExecutor.run(SynchronizedTestExecutor.java:61)
Caused by: java.sql.SQLRecoverableException: Closed Connection
    at oracle.jdbc.driver.PhysicalConnection.prepareStatement(PhysicalConnection.java:3331)
    at oracle.jdbc.driver.T2CConnection.prepareStatement(T2CConnection.java:47)